Fix README.md

johnsimcall opened the pull request:
The previous `oc adm policy add-role-to-user use-scc-hostmount-anyuid ...` command was incorrect.  Replacing it with a one that works for OpenShift 4.6

johnsimcall commented:
I've signed the CLA now.
/assign @jackielii

yonatankahana commented on README.md:
```diff
@@ -61,7 +61,7 @@ $ NAMESPACE=`oc project -q`
 $ sed -i'' "s/namespace:.*/namespace: $NAMESPACE/g" ./deploy/rbac.yaml
 $ oc create -f deploy/rbac.yaml
 $ oc create role use-scc-hostmount-anyuid --verb=use --resource=scc --resource-name=hostmount-anyuid -n $NAMESPACE
```
does this command work for you?

johnsimcall replied:
Yes, the `oc create role...` command completed, but the Provisioner pod was stuck unable to provision storage until I added the SCC to the ServiceAccount.

```
[jcall@jcall-laptop ~]$ export NAMESPACE=nfs-storage

[jcall@jcall-laptop ~]$ oc get role -n $NAMESPACE
NAME                                    CREATED AT
leader-locking-nfs-client-provisioner   2021-03-04T19:03:12Z
use-scc-hostmount-anyuid                2021-03-04T19:03:20Z

[jcall@jcall-laptop ~]$ oc get role/use-scc-hostmount-anyuid -n $NAMESPACE -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: "2021-03-04T19:03:20Z"
  managedFields:
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:rules: {}
    manager: kubectl-create
    operation: Update
    time: "2021-03-04T19:03:20Z"
  name: use-scc-hostmount-anyuid
  namespace: nfs-storage
  resourceVersion: "103774693"
  selfLink: /apis/rbac.authorization.k8s.io/v1/namespaces/nfs-storage/roles/use-scc-hostmount-anyuid
  uid: 0ed79b63-bcc6-45ac-8986-07fab4639758
rules:
- apiGroups:
  - security.openshift.io
  resourceNames:
  - hostmount-anyuid
  resources:
  - securitycontextconstraints
  verbs:
  - use
```

yonatankahana replied:
for me in 3.11 it throws error that i can't use `use` on this resource exactly like #60. 
do you know how to make the readme compatible with both v3 and v4 versions?

jackieli-tes commented:
I'm not familiar with OpenShift, I'll leave to other reviewers to do this. /unassign @jackielii

jackielii commented:
I'm not familiar with OpenShift, I'll leave to other reviewers to do this. /unassign @jackielii

mclosson-lsf commented:
```
$ NAMESPACE=default
$ oc adm policy add-scc-to-user hostmount-anyuid system:serviceaccount:$NAMESPACE:nfs-client-provisioner
scc "hostmount-anyuid" added to: ["system:serviceaccount:default:nfs-client-provisioner"]


$ oc version
oc v3.11.161
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://abcd:8443
openshift v3.11.161
kubernetes v1.11.0+d4cacc0

```

The change works well in my 3.11 cluster.

yonatankahana commented:
> The change works well in my 3.11 cluster.

including `oc create role use-scc-hostmount-anyuid ...`?

yonatankahana commented:
@mclosson-lsf bump

mclosson-lsf commented:
that command fails with this error.

```
$ oc create role use-scc-hostmount-anyuid --verb=use --resource=scc --resource-name=hostmount-anyuid -n $NAMESPACE
error: can not perform 'use' on 'securitycontextconstraints' in group ''
```

mclosson-lsf commented:
I'm not really an expert here.  But, this is the result of my testing on OCP 3.11.

1. it appears that the role is not needed at all.
2. it also appears that access to the hostmount-anyuid scc isn't needed by the service account of the nfs provisioner.


Here's the details for the test for claim number 2.

The nfs-provisioner sa can use the hostmount-anyuid scc
```
$ oc get -oyaml deploy/nfs-client-provisioner | grep -i serviceaccount
      serviceAccount: nfs-client-provisioner
      serviceAccountName: nfs-client-provisioner

$ oc adm policy who-can use scc hostmount-anyuid | grep    nfs-client-provisioner
$ 
```

I run the test-claim and test-pod with an account that can create pvcs, but cannot use the hostmount-anyuid scc.
```
$ oc adm policy who-can create persistentvolumeclaims -ndefault | grep system:serviceaccount:kube-service-catalog:default
        system:serviceaccount:kube-service-catalog:default

$ oc adm policy who-can use scc hostmount-anyuid -ndefault | grep kube-service-catalog
$
```

```
$ oc get po | grep test
test-pod                                  0/1     Completed   0          12m

$ oc get pvc | grep test
test-claim   Bound    pvc-18c3dfff-b1b9-11eb-a73d-6cae8b089043   1Mi        RWX            managed-nfs-storage   13m

$ ls -l /export/nfs-storage/default-test-claim-pvc-18c3dfff-b1b9-11eb-a73d-6cae8b089043/SUCCESS 
-rw-r--r--. 1 1000000000 root 0 May 10 13:57 /export/nfs-storage/default-test-claim-pvc-18c3dfff-b1b9-11eb-a73d-6cae8b089043/SUCCESS
```

yonatankahana commented:
@johnsimcall is this line required in OCP 4.6? if not - can you please update the PR?

johnsimcall commented:
No, the `oc create role ...` command is not required. I've updated my PR so that it is also removed from the README. Thanks to all for the extra set of eyes and confirmation!
